homeassistant/components/vizio/media_player.py

```diff
@@ -84,22 +66,11 @@
         host,
         name,
         token,
-        device_type,
+        VIZIO_DEVICE_CLASSES[device_class],
         session=async_get_clientsession(hass, False),
     )
 
-    if not await device.can_connect():
```

Comment:
Please revert the change regarding this. It's ok to try and improve it in a later PR.

Reply:
> The TV can still be missing from the network, on next start, right? We still need to check that we can connect. We can raise `ConfigEntryNotReady` during entry setup and later if the entity is created we can make it unavailable.
> 
> For this to be effective we need to be able to distinguish between bad credentials and bad connection.

fair point, reverting